### ISSUE TITLE
Object literal ({}).constructor !== Object

### ISSUE THREAD

**Issue by carsonfarmer** [yunabe/tslab#32]
For some reason, when running something like the following in a cell, checks for object literals no longer 'pass'.

```typescript
const blah: { property: string } = { property: 'string'}
console.log(blah.constructor === Object)
console.log(blah.constructor)

// false
// [Function: Object]
```

Versus say:

```typescript
const some = new Object()
console.log(some.constructor === Object)
console.log(some.constructor)

// true
// [Function: Object]
```

Normally, the first 'cell' above should also return true. Any idea why this is the case here? Does it have something to do with how the code is compiled to ES2015 on the fly? This behavior breaks some libraries that rely on object literal checks such as `obj.contructor === Object`. There are workarounds, but this is a pretty unexpected result.

**Comment by yunabe** [comment]
In tslab, we create a new context with `createContext` (in `src/executor.ts`) and run code inside it because we want to hook references of global variables. When the code is executed under the new context, `Object` and `Promise` created for the new context is used (`obj.constructor`).

However, when `Object` in the code is resolved, the reference is forwarded to the Node's original context (aka `global`) and `Object` in the original context is referred. Thus, `obj.constructor` and `Object` point different objects.

This is not very trivial to fix. We need to:
1. Handle references of variables defined in other cells w/o relying on the global scope. We need to change the transpiler.
2. Hook references of global variables w/o creating the new context. I don't feel it's possible with the current Node.js `vm` package.

**Comment by yunabe** [comment]
I released `v1.0.12` with a fix:
https://www.npmjs.com/package/tslab/v/1.0.12
